### PR TITLE
Edit and organize AOT doc

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

The AOT documentation is difficult to follow and inconsistent in style and organization.

Issue Number: https://github.com/angular/angular/issues/17533

## What is the new behavior?

Split the long page into three pages in a new AOT section, edit for style and clarity, rearrange content to introduce and organize related concepts.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/aot-compiler.md [hunk 1]
```diff
@@ -108,26 +92,45 @@ export class TypicalComponent {
...
 
-## Metadata restrictions
+### Compilation phases
+
+There are three phases of AoT compilation.
+* Phase 1 is *code analysis*.
+   In this phase, the TypeScript compiler and  *AoT collector* create a representation of the source. The collector does not attempt to interpret the metadata it collects. It represents the metadata as best it can and records errors when it detects a metadata syntax violation.
+
+* Phase 2 is *code generation*.
+   In this phase, the compiler's `StaticReflector` interprets the `.metadata.json` file created in phase 1, and places restrictions on what it interprets.
```
Comment (IgorMinar) [inline aio/content/guide/aot-compiler.md]:
```suggestion
   In this phase, the compiler's `StaticReflector` interprets the metadata collected in phase 1, performs additional validation of the metadata, and throws an error if it detects a metadata restriction violation.
```

Comment (IgorMinar) [reply aio/content/guide/aot-compiler.md]: metadata.json is used only if you are compiling against libraries, so more generic term "metadata" is better.

File: aio/content/guide/aot-compiler.md [hunk 2]
```diff
@@ -108,26 +92,45 @@ export class TypicalComponent {
...
+
+* Phase 3 is *validation*.
```
Comment (IgorMinar) [inline aio/content/guide/aot-compiler.md]:
```suggestion
* Phase 3 is *template type checking*.
```

File: aio/content/guide/aot-compiler.md [hunk 3]
```diff
@@ -643,612 +555,94 @@ export const ɵ0 = () => new TypicalServer();
...
-
-<!--
-
-Chuck: After reviewing your PR comment I'm still at a loss. See [comment there](https://github.com/angular/angular/pull/17712#discussion_r132025495).
-
--->
 
 {@a binding-expression-validation}
-  ## Phase 3: binding expression validation
+## Phase 3: Binding expression validation
```
Comment (IgorMinar) [inline aio/content/guide/aot-compiler.md]: "template type checking" as I suggested above? in either case the name for this phase should be the same as in the overview above.

File: aio/content/navigation.json [hunk 4]
```diff
@@ -593,15 +593,26 @@
...
+              "title": "Ahead-of-Time Compilation",
+              "tooltip": "Learn why and how to use the Ahead-of-Time (AoT) compiler."
+            },
+            {
+              "url": "guide/angular-compiler-options",
+              "title": "Angular Compiler Options",
+              "tooltip": "Configuring AoT compilation."
+           },
+           {
+              "url": "guide/aot-metadata-errors",
```
Comment (IgorMinar) [inline aio/content/navigation.json]: I like that you separated the errors from the rest. thanks

**Review by IgorMinar — COMMENTED** [review]
I asked @alxhub to review this one, but I did a quick pass through the PR as well.

File: aio/content/guide/aot-compiler.md (hunk 1 above)
Comment (alxhub) [reply aio/content/guide/aot-compiler.md]: 👍

**Review by alxhub — APPROVED** [review]

File: aio/content/guide/aot-compiler.md [hunk 5]
```diff
@@ -1,25 +1,50 @@
-# The Ahead-of-Time (AOT) compiler
+# The Ahead-of-Time (AoT) compiler
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/aot-compiler.md]: No, just wanted to be consistent. I'll change it back.

**Pull request closed without merging**